src/Akka.Persistence.Sql/Query/Dao/BaseByteReadArrayJournalDao.cs

```diff
@@ -52,10 +54,11 @@ public Source<string, NotUsed> AllPersistenceIdsSource(long max)
             var maxTake = MaxTake(max);
 
             return AsyncSource<string>.FromEnumerable(
-                new { _dbStateHolder, maxTake },
+                new { _dbStateHolder, maxTake},
                 static async input =>
                 {
-                    return await input._dbStateHolder.ExecuteWithTransactionAsync(
+                    return await input._dbStateHolder.ExecuteQueryWithTransactionAsync(
```

Comment:
Replace all `ExecuteWithTransactionAsync()` method call inside the read journal with `ExecuteQueryWithTransactionAsync()` to make sure that all DB queries are throttled